src/network.cpp

```diff
@@ -212,80 +212,93 @@ void IRAM_ATTR RemoteLoopEntry(void *)
 }
 #endif
 
-// ConnectToWiFi
-//
-// Connect to the pre-configured WiFi network.
-
 #if ENABLE_WIFI
 
     #define WIFI_WAIT_BASE      4000    // Initial time to wait for WiFi to come up, in ms
     #define WIFI_WAIT_INCREASE  1000    // Increase of WiFi waiting time per cycle, in ms
+    #define WIFI_WAIT_MAX       60000   // Maximum gap between retries, in ms
 
+    #define WIFI_WAIT_INIT      (WIFI_WAIT_BASE - WIFI_WAIT_INCREASE)
 
-    bool ConnectToWiFi(uint cRetries, bool waitForCredentials = false)
+    // ConnectToWiFi
+    //
+    // Try to connect to WiFi using the SSID and password passed as arguments
+    WiFiConnectResult ConnectToWiFi(const String& ssid, const String& password)

```

Comment:
This overload prevents the function's user from having to deal with the oddness of having to pass pointers to the overload that does the actual work.